### PR TITLE
Added keywords that carry over in TPFFactory

### PR DESCRIPTION
**Pull request by christinahedges**
Adds the capability to pass and change keywords in the fits header from TPF factory.

Will also carry forward basic keywords such as MISSION, QUARTER, MODULE, CAMPAIGN etc to the keywords. This should help people distinguish between fits files made from different data/by different teams.

screen shot 2018-03-02 at 5 42 18 pm

### CODE REVIEW

File: lightkurve/targetpixelfile.py [hunk 1]
```diff
@@ -609,15 +609,15 @@ def get_bkg_lightcurve(self, aperture_mask=None):
         return LightCurve(flux=np.nansum(self.flux_bkg[:, aperture_mask], axis=1),
                           time=self.time, flux_err=self.flux_bkg_err)
 
-    def to_fits(self, output_fn=None, overwrite=False):
+    def to_fits(self, output_fn='out.fits', overwrite=False):
```
Comment (barentsen) [inline lightkurve/targetpixelfile.py]: Let's undo this.  There's already a default defined 2 lines below.

File: lightkurve/targetpixelfile.py [hunk 2]
```diff
@@ -729,13 +746,13 @@ def add_cadence(self, frameno, raw_cnts=None, flux=None, flux_err=None,
...
             self.pos_corr2[frameno] = header['POSCORR2']
 
-    def get_tpf(self, **kwargs):
+    def get_tpf(self, keys={}, **kwargs):
         """Returns a KeplerTargetPixelFile object."""
-        return KeplerTargetPixelFile(self._hdulist(), **kwargs)
+        return KeplerTargetPixelFile(self._hdulist(keys=keys), **kwargs)
 
-    def _hdulist(self):
+    def _hdulist(self, keys={}):
```
Comment (barentsen) [inline lightkurve/targetpixelfile.py]: Let's name this `hdu0_keywords` to be explicit about what it does.

**Comment by barentsen** [comment]
@christinahedges Do you want me to review and merge?  Please assign to me if so.

**Comment by christinahedges** [comment]
Not yet..still fiddling.

**Comment by christinahedges** [comment]
@barentsen please can you review this? I've added the ability to pass on keywords and added way more unit tests.

**Comment by barentsen** [comment]
K! Looking into this after lunch. Please long again if  not reviewed within the hour.

**Comment by barentsen** [comment]
This looks really good!  I am happy for this to be merged without changes.

**Pull request merged**